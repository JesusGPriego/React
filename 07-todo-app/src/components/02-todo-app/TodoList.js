import React from 'react'
import { TodoListItem } from './TodoListItem';

export const TodoList = ({props}) => {
    
    // const {todos, handleDelete, handleToggle} = props;
    const [todos, handleDelete, handleToggle] = props;

    
    return (
        <>
            <ul className='list-group list-group-flush'>
                {
                    todos.map((todo,i) => (
                        <TodoListItem key={todo.id} props={[todo, i, handleDelete, handleToggle]}/>
                    ))   
                }
            </ul>
        </>
    )
}
