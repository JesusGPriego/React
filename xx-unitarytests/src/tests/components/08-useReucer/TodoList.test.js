import { shallow } from 'enzyme';
import React from 'react';
import { TodoList } from '../../../components/08-useReducer/TodoList';
import { demoTodo } from '../../fixtures/demoTodos';

describe('testing TodoList Component', () => {
    const handleDelete = jest.fn();
    const handleToggle = jest.fn();

    const wrapper = shallow(
          <TodoList
            todos={demoTodo}
            handleDelete={handleDelete}
            handleToggle={handleToggle}
          />
    )

    

    test('shoul return itself matching snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
    
    test('should display two <TodoListItem />', () => {
    //   console.log(wrapper.find('TodoListItem').length); 
        expect(wrapper.find('TodoListItem').length).toBe(demoTodo.length);
        expect(
            wrapper
            .find('TodoListItem')
            .at(0).prop('handleDelete'))
            .toEqual(
                expect
                .any(
                    Function
                )
            );
    });
    

});
