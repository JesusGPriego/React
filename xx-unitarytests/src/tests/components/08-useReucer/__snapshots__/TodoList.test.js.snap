// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing TodoList Component shoul return itself matching snapshot 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "testing in react",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "learn React",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
